engine/src/main/java/org/entando/entando/aps/servlet/CORSFilter.java

```diff
@@ -0,0 +1,36 @@
+package org.entando.entando.aps.servlet;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ *
+ * @author paddeo
+ */
+public class CORSFilter extends OncePerRequestFilter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        response.addHeader("Access-Control-Allow-Origin", "*");
+
+        if (request.getHeader("Access-Control-Request-Method") != null && "OPTIONS".equals(request.getMethod())) {
+            logger.trace("Sending Header....");
+            // CORS "pre-flight" request
+            response.addHeader("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE");
```

Comment:
the methods should always be included and not only if options is being submitted.
also options should be among the allowed methods.

Reply:
Got it.